juju/juju
# [JUJU-1129] Remove unnecessary abstractions from resources infrastructure

The resources package used too many layers of abstraction.
This PR simplifies the implementation of the resource opener, which fetches a resource from the store and caches in state.

## QA steps

See QA steps for this previous PR

https://github.com/juju/juju/pull/14049

Essentially, deploy charms with resources and run various hook commands.
